tests/tests_geomstats/test_lognormal.py

```diff
@@ -0,0 +1,86 @@
+"""Unit tests for the LogNormal Sampler."""
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.hypersphere import Hypersphere
+from geomstats.geometry.spd_matrices import SPDMatrices
+from geomstats.sampling.lognormal import LogNormal
+
+
+class TestLogNormal(geomstats.tests.TestCase):
+    """Class defining the LogNormal tests."""
+
+    def setUp(self):
+        """Set up  the test"""
+        self.n = 3
+        self.spd_cov_n = (self.n * (self.n + 1)) // 2
+        self.samples = 5
+        self.SPDManifold = SPDMatrices(self.n)
+        self.Euclidean = Euclidean(self.n)
+
+    def test_euclidean_belongs(self):
+        """Test if the samples belong to Euclidean Space"""
+        mean = gs.zeros(self.n)
+        cov = gs.eye(self.n)
+        LogNormalSampler = LogNormal(self.Euclidean, mean, cov)
+        data = LogNormalSampler.sample(self.samples)
+
+        result = gs.all(self.Euclidean.belongs(data))
+        expected = True
+        self.assertAllClose(result, expected)
+
+    def test_spd_belongs(self):
+        """Test if the samples belong to SPD Manifold"""
+        mean = 2 * gs.eye(self.n)
+        cov = gs.eye(self.spd_cov_n)
+        LogNormalSampler = LogNormal(self.SPDManifold, mean, cov)
+        data = LogNormalSampler.sample(self.samples)
+
+        result = gs.all(self.SPDManifold.belongs(data))
+        expected = True
+        self.assertAllClose(result, expected)
+
+    def test_euclidean_frechet_mean(self):
+        """Test if the frechet mean of the samples is close to mean"""
+        mean = gs.zeros(self.n)
+        cov = gs.eye(self.n)
+        data = LogNormal(self.Euclidean, mean, cov).sample(5000)
+        log_data = gs.log(data)
+        fm = gs.mean(log_data, axis=0)
+
+        expected = mean
+        result = fm
+        self.assertAllClose(result, expected, atol=5 * 1e-2)
```

Comment:
Why is atol so low? would `gs.atol` work? (we ask for different precision depending on the backend, hence the `gs.atol`)

Reply:
Here we compute mean from drawn samples and see if it is close to mean of the distribution and so the answer is, we cannot expect more. It would essentially depend on sample efficiency of mean-estimator and all